lib/rules/no-hooks-from-ancestor-modules.js

```diff
@@ -59,11 +59,15 @@ module.exports = {
                 isInModuleCallbackBody(node);
         }
 
-        function isFunctionWithIdentifierArg(node) {
-            return node &&
-                node.type === "FunctionExpression" &&
-                node.params.length > 0 &&
-                node.params[0].type === "Identifier";
+        function getNonThisIdentifierFromParams(params) {
```

Comment:
I feel this would be a bit clearer if we just call this something like `getHooksIdentifierFromParams` (the negative "NonThis" is a little hard to parse at first glance).

Reply:
Good point, renamed.